src/sentry/integrations/slack/event_endpoint.py

```diff
@@ -123,17 +125,16 @@ def on_link_shared(self, request, integration, token, data):
             "unfurls": json.dumps(results),
         }
 
-        session = http.build_session()
-        req = session.post("https://slack.com/api/chat.unfurl", data=payload)
-        status_code = req.status_code
-        response = req.json()
-        track_response_code(status_code, response.get("ok"))
-        req.raise_for_status()
-        if not response.get("ok"):
-            logger.error("slack.event.unfurl-error", extra={"response": response})
+        client = SlackClient()
+        try:
+            client.post("/chat.unfurl", data=payload)
```

Comment:
@MeredithAnya Are we sure that Slack will never send us anything but a `200` status code? Wondering if we could get an `ApiError` from a `4xx` or `500` error.

Reply:
I mean I haven't tested every scenario but I'm sure they send responses other than 200's. So yeah I think the `ApiError` could be from a variety of things, is the problem here that we should be re-raising?

Reply:
@MeredithAnya Yea, we should re-raise it. I noticed that we eat the error in the PD integration but I also know that when an API error happens, it will cause a Sentry error lol: https://sentry.io/organizations/sentry/issues/1551869421/?referrer=jira_integration